src/OrchardCore.Modules/OrchardCore.Demo/Startup.cs

```diff
@@ -102,6 +102,12 @@ public override void ConfigureServices(IServiceCollection services)
                 // Add a custom folder route
                 options.Conventions.AddAreaFolderRoute("OrchardCore.Demo", "/", "Demo");
 
+                 // Add a custom folder route
+                options.Conventions.AddAreaFolderRoute("OrchardCore.Demo", "/Admin", _adminOptions.AdminUrlPrefix + "/Demo");
+
+                // Add a custom page route
+                options.Conventions.AddAreaPageRoute("OrchardCore.Demo", "/OutsideAdmin", _adminOptions.AdminUrlPrefix + "/Demo/OutsideAdmin");
+
```

Comment:
For admin, I am thinking to create `IPageRouteModelConvention` that would remove `/Admin/` and prefix all admin route with `AdminOption` - So need to specify individual route for Admin.

Reply:
Okay good idea, but, as in #5225 for controller actions, we will have to check that for an admin page action there is no more **by default** any valid route without the admin prefix, and that we still allow people to provide routes (with or without the prefix) through modules that will have an higher precedence for link generaration.

For controller actions, the default routes are registered in the `OC.Mvc startup Configure()` that runs last so that people can register routes with an higher precedence. In #5225 this is there that admin actions get other default routes with the admin prefix.